README.md

```diff
@@ -711,7 +711,137 @@ config.customActionDelegate = YourCustomActionDelegate()
+
+##### In App Send
+This event is triggered when an in-app is sent to a user.
+
+Important event properties
+
+- `createdAt` - when the in-app was sent
+- `campaignId`- the in-app campaign id
+
+##### In App Delivery
+This event is triggered when an in-app is received by user on their device.
+
+Important event properties
+
+- `createdAt` - when the in-app was received by the client
+- `campaignId`- the in-app campaign id
+
+
+##### Inbox Message Impression
+This event is triggered when an inbox message is shown in mobile inbox. An impression is registered if the inbox message is displayed in the viewport of mobile inbox. If the inbox message is not visible then an impression is not counted.
+
+Important event properties
+
+- `createdAt` - when the in-app was sent
+- `campaignId`- the in-app campaign id
+- `impressionCount` - how many times the message was in the visible viewport
+- `totalDuration` - the total duration the inbox message was shown in mobile inbox
+
+##### In App Open
+This event is triggered when the user taps on a message in their inbox.
+
+Important event properties
+
+- `createdAt` - when the in-app was opened
+- `campaignId`- the in-app campaign id
+
+
+##### In App Click
+This event is triggered when the user taps on a link or button in an in-app message.
+
+Important event properties
+
+- `createdAt` - when the in-app was sentit
+- `campaignId`- the in-app campaign id
+- `clickedUrl` - the url that was tapped.
+
+##### In App Close
+This event is triggered when the user closes an inbox message by either tapping on 'Close' button or 'Back' button in the navigation controller.
+
+Important event properties
+
+- `createdAt` - when the in-app was closed
+- `campaignId`- the in-app campaign id
+- `closeAction` - it will be `link` if the message was closed due to clicking a link. It will be `back` if the message was closed due to tapping on back button.
+
+##### In App Delete 
+This event is triggered when the user deletes an inbox message by swiping left to delete in mobile inbox or when tapping on 'Delete' button while viewing the inbox message.
+
+Important event properties
+
+- `createdAt` - when the in-app was deleted
+- `campaignId`- the in-app campaign id
+- `deleteAction` - it will be `inbox-swipe` if the delete happened as a result of swiping left on the message in mobile inbox. It will be `delete-button` if the delete happened because 'Delete' button was tapped.
+
+##### Inbox Session
+This event captures an 'inbox session'. An inbox session starts when mobile inbox is shown to the user. An inbox session ends when the user navigates to another part of the application. Please note that inbox session does not end when the user views an inbox message.
+
+Important event properties
+
+- `campaignId`- the in-app campaign id
+- `inboxSessionStart` - when the session started
+- `inboxSessionEnd` - when the session ended
+- `uniqueImpressionCount` - how many unique inbox messages were displayed in the viewport
```

Comment:
It would be good to provide field types for each of these as well here.

Define what an inbox session means? this makes it easier to understand start and end

uniqueImpressionCount - no of unique messages visible in the Inbox message list to the user during the inbox session.

Reply:
Updated the definition `uniqueImpressionCount`.

I am not sure about the field type information here. Seems like too much information for Github readme file. We probably should not give them the impression that this readme file is an exhaustive information regarding inbox events.